spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/StoredProcExecutor.java

```diff
@@ -331,6 +353,20 @@ private String evaluateExpression(Message<?> message) {
 
 	}
 
+	private SimpleJdbcCallOperations obtainSimpleJdbcCall(String storedProcedureName) {
+		if (guavaPresent) {
+			return this.jdbcCallOperationsCache.getUnchecked(storedProcedureName);
+		}
+		else {
+			SimpleJdbcCallOperations operations = this.jdbcCallOperationsMap.get(storedProcedureName);
+			if (operations == null) {
+				operations = createSimpleJdbcCall(storedProcedureName);
+				this.jdbcCallOperationsMap.putIfAbsent(storedProcedureName, operations);
```

Comment:
We need a mechanism to limit the cache size. See #jdbcCallOperationsCacheSize.

Consider using a (synchronized) `LinkedHashMap` (ctor that takes the `accessOrder` boolean). See `LHM.removeEldestEntry()` - the map will take care of limiting the size.